File: python/treatment_planning_data.py
# radiation treatment planning data file
# this one is randomly generated; a real one would use the beam geometry
import scipy.io as sio
import numpy as np

def get_value(mat, name):
    value = mat[name]
    return value[0][0] if np.shape(value) == (1, 1) else value

mat = sio.loadmat("treatment_planning_data.mat")

n = get_value(mat, "n")  # number of beams
mtarget = get_value(mat, "mtarget")  # number of tumor or target voxels
mother = get_value(mat, "mother")  # number of other voxels
Atarget = get_value(mat, "Atarget")
Aother = get_value(mat, "Aother")
Bmax = get_value(mat, "Bmax")
Dtarget = get_value(mat, "Dtarget")
Dother = get_value(mat, "Dother")

File: matlab/treatment_planning_data.m
% radiation treatment planning data file
% this one is randomly generated; a real one would use the beam geometry
rand('state',0);
n = 200; % number of beams
mtarget = 100; % number of tumor or target voxels
mother = 400; % number of other voxels
Atarget = sprand(mtarget,n,0.2);
Atarget = Atarget + [5*sprand(mtarget,mtarget,0.2) zeros(mtarget,n-mtarget)];
Aother= sprand(mother,n,0.2);
Bmax = 10;
Dtarget = 1;
Dother = 0.2;
